src/main/js/field/input/TableSelectInput.js

```diff
@@ -0,0 +1,289 @@
+    constructor(props) {
+        super(props);
+
+        this.retrieveTableData = this.retrieveTableData.bind(this);
+        this.createTable = this.createTable.bind(this);
+        this.createSelect = this.createSelect.bind(this);
+        this.onRowSelected = this.onRowSelected.bind(this);
+        this.onRowSelectedAll = this.onRowSelectedAll.bind(this);
+        this.selectOnClick = this.selectOnClick.bind(this);
+
+        this.state = {
+            progress: false,
+            showTable: false,
+            data: [],
+            selectedData: [],
+            displayedData: []
+        };
+    }
+
+    componentDidMount() {
+        this.state.selectedData.push(...this.props.value);
+        const convertedValues = this.state.selectedData.map(selected => {
+            return Object.assign({ label: selected, value: selected });
+        });
+        this.setState({
+            displayedData: convertedValues
+        })
+    }
+
+    onRowSelectedAll(isSelected, rows) {
+        if (rows) {
+            const selected = Object.assign([], this.state.selectedData);
+            rows.forEach((row) => {
+                this.createSelectedArray(selected, row, isSelected);
+            });
+            this.setState({
+                selectedData: selected
+            });
+        } else {
+            this.setState({
+                selectedData: []
+            });
+        }
+    }
+
+    onRowSelected(row, isSelected) {
+        const selected = Object.assign([], this.state.selectedData);
+        this.createSelectedArray(selected, row, isSelected);
+        this.setState({
+            selectedData: selected
+        });
+    }
+
+    createSelectedArray(selectedArray, row, isSelected) {
+        const keyColumnHeader = this.props.columns.find(column => column.isKey).header;
+        const rowValue = row[keyColumnHeader];
+
+        if (isSelected) {
+            const projectFound = selectedArray.find(project => project === rowValue);
+            if (!projectFound) {
+                selectedArray.push(rowValue);
+            }
+        } else {
+            const index = selectedArray.indexOf(rowValue);
+            if (index >= 0) {
+                selectedArray.splice(index, 1); // if found, remove that element from selected array
+            }
+        }
+    }
+
+    createRowSelectionProps() {
+        return {
+            mode: 'checkbox',
+            clickToSelect: true,
+            showOnlySelected: true,
+            selected: this.state.selectedData,
+            onSelect: this.onRowSelected,
+            onSelectAll: this.onRowSelectedAll
+        };
+    }
+
+    retrieveTableData() {
+        this.setState({
+            progress: true,
+            success: false
+        });
+        const {
+            fieldKey, csrfToken, currentConfig, endpoint
+        } = this.props;
+
+        const request = createNewConfigurationRequest(`/alert${endpoint}/${fieldKey}`, csrfToken, currentConfig);
```

Comment:
Endpoints are always preceded by '/'?

Reply:
The code in Java is expected to pass a url of the form /custom/endpoint/blah/blah which is then inserted here. I could make this friendlier by processing the endpoint url a bit, but figured people often won't be passing their own urls here.